synoptic/src/synoptic/model/export/JsonExporter.java

```diff
@@ -96,151 +110,326 @@ public EventInstance(int traceID, int eventIndexWithinTrace) {
-        assert initialPart != null;
-        if (initialPart == null) {
-            return null;
-        }
+        // add the terminal node last
+        Map<String, Integer> terminalNodeMap = new LinkedHashMap<String, Integer>();
+        terminalNodeMap.put("id", globalID);
+        nodesIDMap.put(evGraph.getTerminalNode(), globalID);
+        terminalNodeMap.put("displayableIDs: ",
+                displayablesIDMap.get("TERMINAL"));
+        nodeList.add(terminalNodeMap);
+        globalID++;
 
-        // Follow all traces and store them in the log list of traces
-        int traceID = 0;
-        for (EventNode startingEvent : initialPart.getEventNodes().iterator()
-                .next().getAllSuccessors()) {
-            // One trace, contains the trace number and a list of events
-            Map<String, Object> singleTraceMap = new LinkedHashMap<String, Object>();
-            // List of events
-            List<Map<String, Object>> singleTraceEventsList = new LinkedList<Map<String, Object>>();
-
-            int eventIndexWithinTrace = 0;
-            for (EventNode event = startingEvent; !event
-                    .isTerminal(); event = event.getAllSuccessors().iterator()
-                            .next()) {
-                // One event, contains event index, event type, and timestamp
-                Map<String, Object> singleEventMap = new LinkedHashMap<String, Object>();
+        return nodeList;
+    }
 
-                // Populate this event's index within the trace, its type, and
-                // the line in the log where it was found
-                singleEventMap.put("eventIndex", eventIndexWithinTrace);
-                EventType evType = event.getEType();
-                singleEventMap.put("eventType", evType.toString());
-                singleEventMap.put("logLine", event.getLineNum());
+    // Map of edges and their respective global ID's
+    static Map<EvBasedEdge, Integer> edgesIDMap = new LinkedHashMap<EvBasedEdge, Integer>();
 
-                // Populate this event's time if it's not INITIAL or TERMINAL
-                if (!evType.isSpecialEventType()) {
-                    singleEventMap.put("timestamp", event.getTime());
+    /**
+     * Creates the 'edges' of the JSON object: a list of edges within this
+     * EvBasedGraph
+     * 
+     * @param evGraph
+     *            The EvBasedGraph whose edges we're outputting
+     */
+    private static List<Map<String, Integer>> makeEdgesJSON(
+            EvBasedGraph evGraph) {
+
+        // List of edges and info to go into the JSON Object
+        List<Map<String, Integer>> edgesList = new LinkedList<Map<String, Integer>>();
+
+        // Get all the nodes in the evGraph
+        Set<EvBasedNode> allNodes = evGraph.getNodes();
+
+        // Contains all edges used to avoid duplicates
+        Set<EvBasedEdge> usedEdges = new HashSet<EvBasedEdge>();
+
+        for (EvBasedNode node : allNodes) {
```

Comment:
Just put `evGraph.getNodes()` directly within this foreach statement, since allNodes isn't used anywhere but here. Make the same change in all other methods that follow this same pattern.